include/dice/template-library/fmt_join.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef DICE_TEMPLATE_LIBRARY_FMT_JOIN_HPP
+#define DICE_TEMPLATE_LIBRARY_FMT_JOIN_HPP
+
+#include <format>
+#include <iterator>
+#include <ranges>
+#include <vector>
+
+namespace dice::template_library {
+
+	namespace fmt_join_algo_detail {
+		template<std::ranges::input_range R, typename Sep>
+		struct joiner {
+			R range;
+			Sep separator;
+		};
+	}// namespace fmt_join_algo_detail
+
+	/**
+	 * @brief Creates a formatting helper alike join from https://fmt.dev/latest/api/#range-and-tuple-formatting
+	 *
+	 * This function is to be used directly inside `std::format`. It does not
+	 * create the final string itself but returns an internal helper object that `std::formatter`
+	 * knows how to process.
+	 *
+	 * @param range The range of elements to join. The elements must be formattable.
+	 * @param separator A string-like object (e.g., ", ") to place between elements.
+	 * @return A helper object for use by `std::format`.
+	 */
+	template<std::ranges::input_range R, typename Sep>
+	constexpr auto fmt_join(R &&range, Sep &&separator) {
+		return fmt_join_algo_detail::joiner{std::views::all(std::forward<R>(range)), std::forward<Sep>(separator)};
+	}
+
+	template<typename T, typename Sep>
+	constexpr auto fmt_join(std::initializer_list<T> range, Sep &&separator) {
+		return fmt_join_algo_detail::joiner{std::views::all(std::vector(range)), std::forward<Sep>(separator)};
```

Comment:
Does initializer list not live long enough?

Reply:
Aparently not. `std::initializer_list` seems to have slipped through a bit for ranges: https://stackoverflow.com/a/76895022